File: app/src/main/java/com/example/schedule/util/LoadDialog.java
package com.example.schedule.util;

import android.os.Bundle;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

import com.example.schedule.R;

public class LoadDialog {

    private LoadDialogFragment loadDialog;
    private FragmentManager fragmentManager;

    public LoadDialog(FragmentManager fragmentManager) {
        this.fragmentManager = fragmentManager;
        loadDialog = new LoadDialogFragment();
    }

    public void show(String text) {
        if (loadDialog == null) return;
        loadDialog.setText(text);
        loadDialog.show(fragmentManager, "loadDialog");
    }

    private Handler changeHandler = new Handler(msg -> {
        if (loadDialog.getDialog() != null)
            ((TextView) loadDialog.getDialog().findViewById(R.id.textView)).setText((String) msg.obj);
        return true;
    });

    public void changeText(String text) {
        changeHandler.sendMessage(changeHandler.obtainMessage(0, text));
    }

    public void close() {
        if (loadDialog != null) loadDialog.dismiss();
    }

    public static class LoadDialogFragment extends DialogFragment {

        String text;

        @Nullable
        @Override
        public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
            View view = inflater.inflate(R.layout.dialog_load, container, false);
            ((TextView) view.findViewById(R.id.textView)).setText(text);
            return view;
        }

        public void setText(String text) {
            this.text = text;
        }
    }

}

File: app/src/main/java/com/example/schedule/util/LoadDialog.kt
package com.example.schedule.util

import android.app.Dialog
import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import com.example.schedule.R

class LoadDialog(private val fragmentManager: FragmentManager) {

    private val loadDialog: LoadDialogFragment = LoadDialogFragment()

    private val handler = Handler()

    fun show(text: String?) {
        loadDialog.text = text
        loadDialog.show(fragmentManager, "loadDialog")
    }

    fun changeText(text: String?) {
        handler.post {
            if (loadDialog.dialog != null)
                loadDialog.dialog!!.findViewById<TextView>(R.id.textView).text = text
        }
    }

    fun close() {
        loadDialog.dismiss()
    }

    class LoadDialogFragment : DialogFragment() {

        var text: String? = null

        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
            val view = inflater.inflate(R.layout.dialog_load, container, false)
            view.findViewById<TextView>(R.id.textView).text = text
            return view
        }

        override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
            val dialog = Dialog(context!!, theme)
            dialog.setTitle("Пожалуйста, подождите")
            return dialog
        }
    }
}